File: JavaIntroNeuralNetworkEdition2/src/com/heatonresearch/book/introneuralnet/ch13/NetworkUtil.java
/**
 * Introduction to Neural Networks with Java, 2nd Edition
 * Copyright 2008 by Heaton Research, Inc. 
 * http://www.heatonresearch.com/books/java-neural-2/
 * 
 * ISBN13: 978-1-60439-008-7  	 
 * ISBN:   1-60439-008-5
 *   
 * This class is released under the:
 * GNU Lesser General Public License (LGPL)
 * http://www.gnu.org/copyleft/lesser.html
 */
package com.heatonresearch.book.introneuralnet.ch13;

import com.heatonresearch.book.introneuralnet.neural.activation.ActivationFunction;
import com.heatonresearch.book.introneuralnet.neural.activation.ActivationSigmoid;
import com.heatonresearch.book.introneuralnet.neural.feedforward.FeedforwardLayer;
import com.heatonresearch.book.introneuralnet.neural.feedforward.FeedforwardNetwork;

/**
 * Chapter 13: Bot Programming and Neural Networks
 * 
 * NetworkUtil: Utility class to create neural networks.
 * 
 * @author Jeff Heaton
 * @version 2.1
 */
public class NetworkUtil {

	public static FeedforwardNetwork createNetwork() {
		final ActivationFunction threshold = new ActivationSigmoid();
		final FeedforwardNetwork network = new FeedforwardNetwork();
		network.addLayer(new FeedforwardLayer(threshold, Config.INPUT_SIZE));
		network.addLayer(new FeedforwardLayer(threshold,
				Config.NEURONS_HIDDEN_1));
		if (Config.NEURONS_HIDDEN_2 > 0) {
			network.addLayer(new FeedforwardLayer(threshold,
					Config.NEURONS_HIDDEN_2));
		}
		network.addLayer(new FeedforwardLayer(threshold, Config.OUTPUT_SIZE));
		network.reset();
		return network;
	}
}

File: CSIntroNeuralNetworkEdition2/Chapter13Bot/NetworkUtil.cs
/// Introduction to Neural Networks with C#, 2nd Edition
/// Copyright 2008 by Heaton Research, Inc. 
/// http://www.heatonresearch.com/book/programming-neural-networks-cs-2.html
/// 
/// ISBN-10: 1604390093
/// ISBN-13: 978-1604390094
/// 
/// This class is released under the:
/// GNU Lesser General Public License (LGPL)
/// http://www.gnu.org/copyleft/lesser.html
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using HeatonResearchNeural.Feedforward;
using HeatonResearchNeural.Activation;

namespace Chapter13Bot
{
    /// <summary>
    /// Utility class to create neural networks.
    /// </summary>
    public class NetworkUtil
    {
        public static FeedforwardNetwork CreateNetwork()
        {
            ActivationFunction threshold = new ActivationSigmoid();
            FeedforwardNetwork network = new FeedforwardNetwork();
            network.AddLayer(new FeedforwardLayer(threshold, Config.INPUT_SIZE));
            network.AddLayer(new FeedforwardLayer(threshold,
                    Config.NEURONS_HIDDEN_1));
            if (Config.NEURONS_HIDDEN_2 > 0)
            {
                network.AddLayer(new FeedforwardLayer(threshold,
                        Config.NEURONS_HIDDEN_2));
            }
            network.AddLayer(new FeedforwardLayer(threshold, Config.OUTPUT_SIZE));
            network.Reset();
            return network;
        }
    }
}
